### DISCUSSION TITLE
Bundle or handling different inputs

### DISCUSSION THREAD

**Discussion by Antonio-Laguna** [johnlindquist/kit#212] [Q&A]
After doing #190 I wanted to be able to use it with the right click with automator. The thing though is that I'd need to then potentially create 2 scripts:

* Hidden one that handles the process of sending a file. Can be tied with automator. 
* Visible one for drag and drop that leverages the above.

Does "add to scriptkit" support this scenario? Or better, do we have a way to check if the script is being launched from shell or from the app? If that'd be the case I could do:

```js
let input;

if (FROM_APP) {
  input = await drop('Drop an image')[0];
} else {
  input = await arg();
}
```

And just handle the rest.

Thanks a lot for this tool John! Looks awesome!

Also, do you have any CHANGELOG somewhere? I often see the app updating and always wonder what's the new thing I can play with.

**Comment by johnlindquist** [comment]
`env.KIT_CONTEXT` will be "app" if you run it from the app.

I'm planning on adding more like, "kar", "terminal", "server", etc, but "app" is the only one I've set up so far. 
(I've got a few related things to sort out like passing ENV vars and arg --flags from other contexts).

> 📝 Note: It's also encouraged to use `edit->duplicate` in the main menu rather than trying to make scripts with too many conditions or too much logic.

No CHANGELOG so far. The plan is to put announcements, etc in the newsletter (the sign-up is in the `join` in the main menu) and the [announcements](https://github.com/johnlindquist/kit/discussions/categories/announcements) category in discussions. In fact, I need to go post about a new feature in there right now 😇

**Reply by Antonio-Laguna** [reply]
Looks like `kar` stills sets the `KIT_CONTEXT` to `app`

**Reply by johnlindquist** [reply]
That's expected. `kar` sends a message to the app to run the specified script. So it's still the App that launches the script.

Using `kar` means:
- App prompts with launch
- You have access to `await drop()` and other app-specific utils (I need to make a list)
- If a prompt is already open, it will cancel the previous script to start the new one. The App only allows one script at a time since user-input is expected and we don't want any runaway processes.

Using `script` means:
- App doesn't launch
- Run as many scripts as you want. But runaway scripts will need to be killed by process ids.

I'll figure it all out so that no matter where/how you run a script, you have Environment variables telling you what's going on. But again, duplicating scripts for specific purposes is always encouraged.

**Reply by Antonio-Laguna** [reply]
The thing though is that it'd be duplicating all the logic instead of just forking where the input comes from? If you still think that's the way to go I'll do that!
